### PR TITLE
task 9355: recovery wallet screen: create ui

### PR DESCRIPTION
**Pull request by vanpd-0859**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9355
### Goals :soccer:

https://marvelapp.com/3feje8d/screen/54838597
- Segment Control with 2 tabs: Mnenomic Phrase and Private Key
- Note label
- Mnenomic or Private Key textview
- Set password label
- Password + repeat password textfield
- Recover Wallet button
- What is a mnenomic phrase or What is a private key button

### Implementation Details :construction:

Screen Shot 2019-04-04 at 10 20 51 AM
Screen Shot 2019-04-04 at 10 22 01 AM

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift [hunk 1]
```diff
@@ -8,13 +8,80 @@
...
+            textView.do {
+                $0.text = nil
+                $0.textColor = .black
+            }
+        }
+    }
+    
+    func textViewDidEndEditing(_ textView: UITextView) {
+        if textView.text.isEmpty {
+            textView.text = recoverDataPlaceHolder
```
Comment (dongsonbk58) [inline CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: dùng do

**Review by dongsonbk58 — APPROVED** [review]

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift [hunk 2]
```diff
@@ -8,13 +8,80 @@
...
+    }
+    
+    @IBAction private func handleGuideTapped(_ sender: Any) {
+    }
+    
+    @IBAction func handleRecoverMethodTabChanged(_ sender: UISegmentedControl) {
+        let methodName = sender.titleForSegment(at: sender.selectedSegmentIndex) ?? "Mnenomic Phrase"
+        recoverWalletNoticeLabel.text = "You can reset the password while importing the \(methodName)."
+        recoverDataPlaceHolder = methodName
+        if recoverDataTextView.textColor == .lightGray {
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: Sử dụng uitextview placeholder
https://github.com/devxoul/UITextView-Placeholder

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift [hunk 3]
```diff
@@ -8,13 +8,80 @@
...
+        }
+        recoverDataTextView.do {
+            $0.delegate = self
+            $0.textColor = .lightGray
+            $0.text = recoverDataPlaceHolder
+        }
     }
 }
 
 extension RecoverWalletViewController: StoryboardSceneBased {
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: Nên cho xuống dưới cùng

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift [hunk 4]
```diff
@@ -8,13 +8,80 @@
...
     static var sceneStoryboard = Storyboards.main
 }
+
+extension RecoverWalletViewController: UITextViewDelegate {
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: Sử dụng placeholder

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift (hunk 2 above)
Comment (vanpd-0859) [reply CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: Anh ơi, uitextview làm gì có thuộc tính placeholder anh nhỉ? chỉ có textfield mới có chứ?

**Comment by vanpd-0859** [comment]
ready

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**